h2o-py/tests/testdir_jira/pyunit_pubdev_7252_expose_original_training_frame.py

```diff
@@ -0,0 +1,61 @@
+    prostate[1] = prostate[1].asfactor()
+    prostate.summary()
+    ntrees_original = 1000
+    encoding = "one_hot_explicit"
+
+
+    prostate_gbm_enc = H2OGradientBoostingEstimator(categorical_encoding=encoding, nfolds=5,ntrees=ntrees_original, distribution="bernoulli", stopping_metric="MSE", stopping_tolerance=0.01, stopping_rounds=5, seed=1, score_tree_interval = 10)
+    prostate_gbm = H2OGradientBoostingEstimator(nfolds=5,ntrees=ntrees_original, distribution="bernoulli", stopping_metric="MSE", stopping_tolerance=0.01, stopping_rounds=5, seed=1, score_tree_interval = 10)
+    
+    prostate_gbm_enc.train(x=list(range(2,9)), y=1, training_frame=prostate)
+    prostate_gbm.train(x=list(range(2,9)), y=1, training_frame=prostate)
+
+    assert(prostate_gbm._model_json['output']['names'] == prostate_gbm_enc._model_json['output']['original_names'])
+
+
+    prostate_if_enc = H2OIsolationForestEstimator(categorical_encoding=encoding, sample_rate = 0.1, max_depth = 20, ntrees=ntrees_original, stopping_metric="anomalyscore", stopping_tolerance=0.01, stopping_rounds=5, seed=1, score_tree_interval = 10)
+    prostate_if = H2OIsolationForestEstimator(sample_rate = 0.1, max_depth = 20, ntrees=ntrees_original, stopping_metric="anomalyscore", stopping_tolerance=0.01, stopping_rounds=5, seed=1, score_tree_interval = 10)
+
+    prostate_if_enc.train(x=list(range(2,9)), y=1, training_frame=prostate)
+    prostate_if.train(x=list(range(2,9)), y=1, training_frame=prostate)
+
+    assert(prostate_if._model_json['output']['names'] == prostate_if_enc._model_json['output']['original_names'])
+
+
+    prostate_drf_enc = H2ORandomForestEstimator(categorical_encoding=encoding, ntrees=ntrees_original, max_depth=20, min_rows=10, stopping_metric="auc", stopping_tolerance=0.01, stopping_rounds=5, seed=1, score_tree_interval = 10)
+    prostate_drf = H2ORandomForestEstimator(ntrees=ntrees_original, max_depth=20, min_rows=10, stopping_metric="auc", stopping_tolerance=0.01, stopping_rounds=5, seed=1, score_tree_interval = 10)
+
+    prostate_drf_enc.train(x=list(range(2,9)), y=1, training_frame=prostate)
+    prostate_drf.train(x=list(range(2,9)), y=1, training_frame=prostate)
+
+    assert(prostate_drf._model_json['output']['names'] == prostate_drf_enc._model_json['output']['original_names'])
+
+
+    prostate_xgb_enc = H2OXGBoostEstimator(categorical_encoding=encoding, distribution="auto", ntrees=ntrees_original, seed=1, stopping_metric="auc", stopping_tolerance=0.01, stopping_rounds=5, score_tree_interval = 10)
+    prostate_xgb = H2OXGBoostEstimator(distribution="auto", ntrees=ntrees_original, seed=1, stopping_metric="auc", stopping_tolerance=0.01, stopping_rounds=5, score_tree_interval = 10)
+
+
+    prostate_xgb_enc.train(x=list(range(2,9)), y=1, training_frame=prostate)
+    prostate_xgb.train(x=list(range(2,9)), y=1, training_frame=prostate)
+
+    assert(prostate_drf._model_json['output']['names'] == prostate_drf_enc._model_json['output']['original_names'])
```

Comment:
should be prostate_xgb

Reply:
Thanks, fixed..  Hm this fix revealed that the behaviour is inconsistent: for XGBoostEstimator prostate.names == prostate_xgb_enc._model_json['output']['original_names']
but prostate_xgb._model_json['output']['names'] != prostate_xgb_enc._model_json['output']['original_names']. For others, prostate_xy._model_json['output']['names'] == prostate_xy_enc._model_json['output']['original_names'], but prostate.names != prostate_xy_enc._model_json['output']['original_names']

@laurendiperna can you please help here - What is the required behaviour here?

Reply:
This needs to be tested on categorical data. Will rewrite the test.